types/src/contracts.rs

```diff
@@ -170,6 +259,295 @@ pub type DisabledVersions = BTreeSet<ContractVersionKey>;
+    }
+}
+
+impl ToBytes for ContractHash {
+    #[inline(always)]
+    fn to_bytes(&self) -> Result<Vec<u8>, bytesrepr::Error> {
+        self.0.to_bytes()
+    }
+
+    #[inline(always)]
+    fn serialized_length(&self) -> usize {
+        self.0.serialized_length()
+    }
+}
+
+impl FromBytes for ContractHash {
+    fn from_bytes(bytes: &[u8]) -> Result<(Self, &[u8]), bytesrepr::Error> {
+        let (bytes, rem) = FromBytes::from_bytes(bytes)?;
+        Ok((ContractHash::new(bytes), rem))
+    }
+}
+
+impl From<[u8; 32]> for ContractHash {
+    fn from(bytes: [u8; 32]) -> Self {
+        ContractHash(bytes)
+    }
+}
+
+impl Serialize for ContractHash {
+    fn serialize<S: Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
+        if serializer.is_human_readable() {
+            self.to_formatted_string().serialize(serializer)
+        } else {
+            self.0.serialize(serializer)
+        }
+    }
+}
+
+impl<'de> Deserialize<'de> for ContractHash {
+    fn deserialize<D: Deserializer<'de>>(deserializer: D) -> Result<Self, D::Error> {
+        if deserializer.is_human_readable() {
+            let formatted_string = String::deserialize(deserializer)?;
+            ContractHash::from_formatted_str(&formatted_string).map_err(SerdeError::custom)
+        } else {
+            let bytes = HashAddr::deserialize(deserializer)?;
+            Ok(ContractHash(bytes))
+        }
+    }
+}
+
+impl AsRef<[u8]> for ContractHash {
+    fn as_ref(&self) -> &[u8] {
+        self.0.as_ref()
+    }
+}
+
+impl TryFrom<&[u8]> for ContractHash {
+    type Error = TryFromSliceForContractHashError;
+
+    fn try_from(bytes: &[u8]) -> Result<Self, TryFromSliceForContractHashError> {
+        HashAddr::try_from(bytes)
+            .map(ContractHash::new)
+            .map_err(|_| TryFromSliceForContractHashError(()))
+    }
+}
+
+impl TryFrom<&Vec<u8>> for ContractHash {
+    type Error = TryFromSliceForContractHashError;
+
+    fn try_from(bytes: &Vec<u8>) -> Result<Self, Self::Error> {
+        HashAddr::try_from(bytes as &[u8])
+            .map(ContractHash::new)
+            .map_err(|_| TryFromSliceForContractHashError(()))
+    }
+}
+
+#[cfg(feature = "std")]
+impl JsonSchema for ContractHash {
+    fn schema_name() -> String {
+        String::from("ContractHash")
+    }
+
+    fn json_schema(gen: &mut SchemaGenerator) -> Schema {
+        let schema = gen.subschema_for::<String>();
+        let mut schema_object = schema.into_object();
+        schema_object.metadata().description = Some("The hash address of the contract".to_string());
+        schema_object.into()
+    }
+}
+
+/// A newtype wrapping a `HashAddr` which is the raw bytes of
+/// the ContractPackageHash
+#[derive(DataSize, Default, PartialOrd, Ord, PartialEq, Eq, Hash, Clone, Copy)]
+pub struct ContractPackageHash(HashAddr);
+
+impl ContractPackageHash {
+    /// Constructs a new `ContractHash` from the raw bytes of the contract hash.
```

Comment:
```suggestion
    /// Constructs a new `ContractPackageHash` from the raw bytes of the contract hash.
```

You forgot to refactor all "contract hash" comments to "contract package hash".

Reply:
Applied suggestion